### ISSUE TITLE
Compiling errors

### ISSUE THREAD

**Issue by tomberek** [svanderburg/dydisnix#1]
From scratch on NixOS with autoconf,automake,libtool:
```
# output of ./bootstrap
autoreconf: Entering directory `.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: aclocal --force
autoreconf: configure.ac: tracing
autoreconf: configure.ac: not using Libtool
autoreconf: running: /nix/store/rdnli6k94jlgjpyf8ccr9hgdp3pi4p01-autoconf-2.69/bin/autoconf --force
configure.ac:18: error: possibly undefined macro: AC_PROG_LIBTOOL
      If this token and others are legitimate, please use m4_pattern_allow.
      See the Autoconf documentation.
autoreconf: /nix/store/rdnli6k94jlgjpyf8ccr9hgdp3pi4p01-autoconf-2.69/bin/autoconf failed with exit status: 1
```

On OSX: 
```
# output of ./configure
./configure: line 4: .: filename argument required
.: usage: . filename [arguments]
```

This is solved by editing line 4 to change '.' to '#'. But then we still have the following:
```
checking for a BSD-compatible install... /usr/local/bin/ginstall -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /usr/local/bin/gmkdir -p
checking for gawk... no
checking for mawk... no
checking for nawk... no
checking for awk... awk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking how to create a pax tar archive... gnutar
checking for gcc... gcc
checking whether the C compiler works... no
configure: error: in `/Users/tomberek/Documents/IMOS-nix/dydisnix':
configure: error: C compiler cannot create executables
See `config.log' for more details
```

**Comment by svanderburg** [comment]
Looks like there are a couple of things wrong with your installation. Since you have been using an extension to Disnix (which itself is an extension to Nix), can you try building it with the `release.nix` expression included with the package?

```bash
$ nix-env -f release.nix -iA build.x86_64-linux
```

This approach is probably more convenient.

**Comment by tomberek** [comment]
Yes, that has worked well. Thanks. The one issue I ran into is that I am using OSX with a NixOS build slave.

```
-    services.disnixTest.package = mkDefault (import ./release.nix {}).build."${builtins.currentSystem}";
+    services.disnixTest.package = mkDefault (import ./release.nix {}).build."${builtins.targetSystem or builtins.buildSystem}";
```
Those builtins don't exist, but that's the gist of what might be better.

I was able to build the module by including darwin as an acceptable value for `system` in release.nix and then editing the module afterward with x86_64-linux for the slave to be able to build it.
